third_party/clang/iOS.cmake

```diff
@@ -104,6 +108,9 @@ IF (NOT DEFINED IOS_MIN_TARGET)
 ENDIF()
 
 SET(IOS_COMMON_FLAGS "-m${PLATFORM_FLAG_NAME}-version-min=${IOS_MIN_TARGET}")
+if (CATALYST)
+  SET(IOS_COMMON_FLAGS "${IOS_COMMON_FLAGS} -target ${IOS_ARCH}-apple-ios${IOS_MIN_TARGET}-macabi")
+endif()
```

Comment:
At first glance I wasn't too happy about patching this toolchain file, which comes from [opensource.apple.com](https://opensource.apple.com/source/clang/clang-800.0.42.1/src/cmake/platforms/iOS.cmake.auto.html), but I see this has been the practice before upstream. It's definitely not our concern right now (or at all), but https://github.com/leetal/ios-cmake could be a useful replacement, with built-in support for Universal libs, Catalyst and Apple Silicon.